### PR TITLE
Add more component tests

### PR DESCRIPTION
**Pull request by dashaxiong**
This PR includes more unit tests for app/js/components. It helps with issue #368. 

The tests are based on existing code, testing some of the ways that components are rendered when supplied with different props. They typically check if the render function produces particular elements with specific attributes and text nodes. Very happy to discuss ways in which the tests may be improved :hammer: :grin:  

New test suites are added for components that were previously untested.

### CODE REVIEW

File: test/components/Navigation.test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }  
+  
+    const wrapper = shallow(<Navigation {...props} />)
+
+    it('uses a ChevronLeftIcon', () => {
+      expect(wrapper.find('ChevronLeftIcon').exists()).to.equal(true)
+    })
+
+    it('includes the previous label', () => {
+      expect(wrapper.childAt(0).childAt(1).text()).to.contain('Back')
```
Comment (wbobeirne) [inline test/components/Navigation.test.js]: This is really brittle, selecting specific child indices. It should probably search for an element by tag, or simply check that the text is contained.

File: test/components/Navigation.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  describe('renders the component with a label', () => {
+    const props = {
+      previous() {},
+      previousLabel: 'Previous Page'
+    }
+
+    const wrapper = shallow(<Navigation {...props} />)
+
+    it('includes the supplied label', () => {
+      expect(wrapper.childAt(0).childAt(1).text()).to.contain('Previous Page')
```
Comment (wbobeirne) [inline test/components/Navigation.test.js]: Same as above.

File: test/components/Navbar.test.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    })
+
+    it('uses the regular wallet tab icon', () => {
+      expect(wrapper.find('img[src="/images/icon-nav-wallet.svg"]').exists()).to.equal(true)
+    })
+
+    it('uses the active profile tab icon', () => {
+      expect(wrapper.find('img[src="/images/icon-nav-profile-hover.svg"]').exists()).to.equal(true)
+    })
+  })
```
Comment (wbobeirne) [inline test/components/Navbar.test.js]: I'd like to see less copy-paste of these tests. Looping over an array of possibilities would make for a more easily maintained test if something about Navbar changes.

Comment (dashaxiong) [reply test/components/Navbar.test.js]: Thanks for your feedback, I agree. I will attempt those changes today.

**Pull request merged**